Assets/Scripts/Controller.cs

```diff
@@ -22,11 +22,26 @@ void Update()
 	{
 		if (controller.triggerPressed)
 		{
-			laser.transform.localScale = new Vector3(2, 1, 2);
+			laser.transform.localScale = new Vector3(2, laser.transform.localScale.y, 2);
 		}
 		else
 		{
-			laser.transform.localScale = new Vector3(1, 1, 1);
+			laser.transform.localScale = new Vector3(1, laser.transform.localScale.y, 1);
+		}
+
+		var ray = new Ray(controller.transform.position, controller.transform.forward);
+		RaycastHit hit;
+		Physics.Raycast(ray, out hit, 100, 1 << LayerMask.NameToLayer("Seekbar"));
+
+		if (hit.transform != null)
```

Comment:
What does this block do?

Reply:
It shortens the laser when you are aiming at the Seekbar

Reply:
Could you use RaycastHit.distance instead of hardcoding it?

Reply:
Could potentially also eliminate the if(), if RaycastHit.distance gives a sensible value if not hitting anything.

Reply:
It simply returns 0 when it doesn't hit a collider, but it does work for setting the laser length when it hits the Seekbar.